### PR TITLE
Tighten compliance rules for LLP companies

### PR DESCRIPTION
**Pull request by justinblandford1**
This pull request introduces stricter validation and improved error handling for officer roles in LLP company types, ensures appropriate defaulting logic for appointments, and adds comprehensive test coverage for these scenarios. It also enhances exception handling to provide clearer client feedback and ensures proper rollback on failures.

**Validation and Business Logic Enhancements:**
- LLP officer roles are now strictly validated to ensure only valid roles are accepted for LLP company types, and vice versa. Invalid combinations result in a `400 Bad Request` with a clear error message. (`AppointmentsServiceImpl.java`, `ApiExceptionHandler.java`) [[1]](diffhunk://#diff-2ab3a21436f958096c1df5058580519742e649d54d0b0abd7e373ba441951f0aR108-R137) [[2]](diffhunk://#diff-afb54b1f95a338c49b5533150e10d15c9954f52f26de5424aa5f8e3f83deeea5R58-R65)
- Defaulting logic for the number of appointments and officer roles is updated: LLPs default to two designated members if not explicitly set, and the number of appointments for LLPs is capped at 20. (`AppointmentsServiceImpl.java`) [[1]](diffhunk://#diff-2ab3a21436f958096c1df5058580519742e649d54d0b0abd7e373ba441951f0aR56) [[2]](diffhunk://#diff-2ab3a21436f958096c1df5058580519742e649d54d0b0abd7e373ba441951f0aR84-R96) [[3]](diffhunk://#diff-2ab3a21436f958096c1df5058580519742e649d54d0b0abd7e373ba441951f0aR108-R137)

**Error Handling Improvements:**
- `IllegalArgumentException`s thrown due to invalid officer roles are now handled by returning a structured validation error response with HTTP 400 status. (`ApiExceptionHandler.java`, `ApiExceptionHandlerTest.java`) [[1]](diffhunk://#diff-afb54b1f95a338c49b5533150e10d15c9954f52f26de5424aa5f8e3f83deeea5R58-R65) [[2]](diffhunk://#diff-ac0823b72db25f885f890482b1186e034d04228e7c73c1c9264b917f1664eb2fR196-R217)
- The company creation workflow now attempts a rollback (delete) on both general and validation failures, logging any rollback failures for better traceability. (`CreateCompanyWorkflowServiceImpl.java`) [[1]](diffhunk://#diff-76e4ebad41cd70421a265bc27b92c1f16dd27cb0f465f741a0acfc97021a3697R285-R287) [[2]](diffhunk://#diff-76e4ebad41cd70421a265bc27b92c1f16dd27cb0f465f741a0acfc97021a3697R320-L322)

**Test Coverage and Backwards Compatibility:**
- New tests ensure that legacy V1 endpoints still accept LLP officer payloads that V2 endpoints reject, preserving backwards compatibility. (`InternalCompanyControllerTest.java`, `PublicCompanyControllerTest.java`, `InternalCompanyRequestV2ArgumentResolverTest.java`) [[1]](diffhunk://#diff-b3d545789c558975a3796fa3fbf455a00ba8b472d62fa5d92ca3d4eafd56d0a2R115-R136) [[2]](diffhunk://#diff-1b6d24770b3150025d02ffe80cfc7a02e713397cfa152ec9855890797ecad1b6R105-R126) [[3]](diffhunk://#diff-278281142ae23bd1c20e45fe5ac8c7041330c8ed88c245502d09c750859f24dbR79-R95)
- Additional tests verify proper exception handling and error responses for invalid officer roles. (`ApiExceptionHandlerTest.java`)

These changes collectively enforce stricter data integrity for LLP appointments and provide clearer, more actionable feedback to API clients.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by lyadavalli — DISMISSED** [review]

File: src/main/java/uk/gov/companieshouse/api/testdata/service/impl/AppointmentsServiceImpl.java [hunk 1]
```diff
@@ -385,6 +287,215 @@ private void createSingleAppointment(
...
+        long designatedMembers = officerRoleList.stream()
+                .filter(role -> role == OfficerType.LLP_DESIGNATED_MEMBER)
+                .count();
+        while (designatedMembers < DEFAULT_LLP_APPOINTMENTS) {
+            officerRoleList.add(0, OfficerType.LLP_DESIGNATED_MEMBER);
+            designatedMembers++;
+        }
+
+        int adjustedNumberOfAppointments = Math.max(numberOfAppointments, officerRoleList.size());
+        if (adjustedNumberOfAppointments > 20) {
```
Comment (kuzonduch) [inline src/main/java/uk/gov/companieshouse/api/testdata/service/impl/AppointmentsServiceImpl.java:368]: The 20limit check on line 368 fires after the mandatory designated members have already been prepended (lines 362–364). This means a caller supplying 19 LLP_MEMBER roles — which is within the stated limit — will receive a misleading "Number of appointments must not exceed 20" error.
The effective user-supplied limit is 18, not 20.

Moving the cap check to before the while loop or Updating the error message to reflect the actual constraint

There's also no test covering this 18/19-role boundary case.

**Review by tlawrencech — DISMISSED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by tlawrencech — APPROVED** [review]

**Review by bsrinivasancompanieshouse — APPROVED** [review]

**Review by lyadavalli — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/api/testdata/service/impl/AppointmentsServiceImpl.java (hunk 1 above)
Comment (justinblandford1) [reply src/main/java/uk/gov/companieshouse/api/testdata/service/impl/AppointmentsServiceImpl.java:368]: done
